CMakeLists.txt

```diff
@@ -54,6 +54,10 @@ else()
   message("Doing a local build")
 endif()
 
+if($ENV{NUPIC_ARM_BUILD})
```

Comment:
Yes, why do we need the `NUPIC_ARM_BUILD` environment variable? The proposed use of [CMAKE_SYSTEM_PROCESSOR](https://cmake.org/cmake/help/v3.0/variable/CMAKE_SYSTEM_PROCESSOR.html) seems to simplify the build, and also eliminates the need to define, document, and maintain an unnecessary additional variable `PLATFORM_TYPE`.

The variable `PLATFORM_TYPE` is particularly problematic, as it's only set for the arm CPU type ("platform" is a pretty generic term, anyway). Since `PLATFORM_TYPE` is available to all other scripts, it becomes error-prone and confusing, as it doesn't reflect other CPU types, which other scripts might expect.

Thus, `CMAKE_SYSTEM_PROCESSOR` obviates `PLATFORM_TYPE`.